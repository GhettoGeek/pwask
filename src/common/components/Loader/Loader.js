import React from 'react'
import PropTypes from 'prop-types'
import { CircularProgress } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'

const useStyles = makeStyles((theme) => ({
  root: {
    margin: theme.spacing(2),
  },
}))

function Loader({ size }) {
  const classes = useStyles()

  return (
    <CircularProgress size={size} className={classes.root} />
  )
}

Loader.propTypes = {
  size: PropTypes.number,
}

Loader.defaultProps = {
  size: 40,
}

export default Loader
