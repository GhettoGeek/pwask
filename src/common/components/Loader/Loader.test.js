import React from 'react'
import { shallow } from 'enzyme'
import Loader from './Loader'

describe('Loader', () => {
  it('renders correctly the component', () => {
    const wrapper = shallow(<Loader />)

    expect(wrapper.html()).toMatchSnapshot()
  })
})
