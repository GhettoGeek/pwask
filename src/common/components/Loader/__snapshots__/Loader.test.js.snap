// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders correctly the component 1`] = `"<div class=\\"MuiCircularProgress-root makeStyles-root-1 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate\\" style=\\"width:40px;height:40px\\" role=\\"progressbar\\"><svg class=\\"MuiCircularProgress-svg\\" viewBox=\\"22 22 44 44\\"><circle class=\\"MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate\\" cx=\\"44\\" cy=\\"44\\" r=\\"20.2\\" fill=\\"none\\" stroke-width=\\"3.6\\"></circle></svg></div>"`;
